chatsky/llm/prompt.py

```diff
@@ -40,12 +43,179 @@ class Prompt(BaseModel):
+
+    async def to_langchain_messages(
+        self, 
+        ctx: Context, 
+        source: str = "human",
+        position_config: Optional[PositionConfig] = None
+    ) -> List[Union[HumanMessage, SystemMessage, AIMessage]]:
+        """
+        Convert this prompt to a list of Langchain messages.
+
+        :param ctx: Current dialog context.
+        :param source: Source of the message ["human", "ai", "system"]. Defaults to "human".
+        :param position_config: Configuration for positioning different parts of the context.
+            If None, will use default PositionConfig().
+        :return: List of Langchain message objects.
+        """
+        message = await self.message(ctx)
+        if message.text == "":
+            return []
+            
+        langchain_message = await message_to_langchain(message, ctx, source=source)
+        return [langchain_message]
+
+
+class Prompt(BasePrompt):
+    """
+    LLM Prompt.
+    Provides position config and allow validating prompts from message or string.
+    
+    This class implements BasePrompt interface and provides concrete implementation
+    for converting various message types (string, Message, BaseResponse) into a Message object.
+    """
+
+    message: AnyResponse
+    """
+    The message content of the prompt. Can be:
+    - string: will be converted to Message with text field
+    - Message: will be used as is
+    - BaseResponse: will be evaluated in the context to get Message
+    """
+
     def __init__(self, message: Union[MessageInitTypes, BaseResponse], position: Optional[float] = None):
+        """
+        Initialize a new Prompt.
+
+        :param message: The message content for this prompt
+        :param position: Optional position in the context sequence
+        """
         super().__init__(message=message, position=position)
 
     @model_validator(mode="before")
     @classmethod
     def validate_from_message(cls, data):
+        """
+        Validate and convert input data to proper format.
+        Allows creating Prompt from string, Message or BaseResponse directly.
+
+        :param data: Input data to validate
+        :return: Validated data in proper format
+        """
         if isinstance(data, (str, Message, BaseResponse)):
             return {"message": data}
         return data
+
+    async def message(self, ctx: Context) -> Message:
+        """
+        Get the message content for this prompt.
+        Implements the abstract method from BasePrompt.
+
+        :param ctx: Current dialog context
+        :return: Message object containing the prompt content
+        :raises ValueError: If message cannot be converted to proper format
+        """
+        try:
+            if isinstance(self.message, Message):
+                return self.message
+            elif isinstance(self.message, BaseResponse):
+                return await self.message(ctx)
+            else:
+                return Message(text=str(self.message))
```

Comment:
Is it possible for `self.message` to be of class `Message` according to [`AnyResponse` type annotation](https://github.com/deeppavlov/chatsky/blob/master/chatsky/core/script_function.py#L158)?